node/src/peer_tracker.rs

```diff
@@ -127,26 +135,30 @@ impl PeerTracker {
             }
         }
 
-        peer_info.state = PeerState::Connected;
+        // This is needed to avoid downgrading `Identified` state to `Connected`
+        if !peer_info.is_connected() {
+            peer_info.state = PeerState::Connected;
+        }
+
         peer_info.connections.push(connection_id);
 
         // If this is the first connection from the peer
         if peer_info.connections.len() == 1 {
-            increase_connected_peers(&self.info_tx, peer_info.trusted);
+            increment_connected_peers(&self.info_tx, peer_info.trusted);
```

Comment:
I think this could be merged into the previous `if`

Reply:
done